src/gen_rules.ml

```diff
@@ -1083,11 +1084,11 @@ let gen ~contexts ~build_system
     let module M = Gen(struct let sctx = sctx end) in
     Fiber.Ivar.fill (Option.value_exn (Hashtbl.find sctxs context.name)) sctx
     >>| fun () ->
-    (context.name, ((module M : Gen), stanzas))
+    (context.name, (module M : Gen))
   in
   Fiber.parallel_map contexts ~f:make_sctx >>| fun l ->
   let map = String.Map.of_list_exn l in
   Build_system.set_rule_generators build_system
-    (String.Map.map map ~f:(fun ((module M : Gen), _) -> M.gen_rules));
-  String.Map.iter map ~f:(fun ((module M : Gen), _) -> M.init ());
-  String.Map.map map ~f:snd
+    (String_map.map map ~f:(fun (module M : Gen) -> M.gen_rules));
+  String_map.iter map ~f:(fun (module M : Gen) -> M.init ());
+  String_map.map map ~f:(fun (module M : Gen) -> M.sctx)
```

Comment:
This is correct, basically for the `printenv` command we need to access the super context. So instead of keeping the stanzas, we just keep the whole super contexts now.